nemo_text_processing/g2p/modules.py

```diff
@@ -496,3 +496,95 @@ def __call__(self, text):
             prons.extend(pron)
 
         return prons
+
+
+class ChineseG2p(BaseG2p):
+    def __init__(
+        self,
+        phoneme_dict=None,
+        word_tokenize_func=None,
+        apply_to_oov_word=None,
+        ignore_ambiguous_words=False,
+        mapping_file: Optional[str] = None,
+    ):
+        """Chinese G2P module. This module first converts Chinese characters into pinyin sequences using pypinyin, then pinyin sequences would 
+           be further converted into phoneme sequences using pinyin2phonemes dict file. For Chinese and English bilingual sentences, the English words
+           would be converted into letters.
+        Args:
+            phoneme_dict (str, Path, Dict): Path to pinyin2phonemes dict file.
+            word_tokenize_func: Function for tokenizing text to words.
+                It has to return List[Tuple[Union[str, List[str]], bool]] where every tuple denotes word representation and flag whether to leave unchanged or not.
+                It is expected that unchangeable word representation will be represented as List[str], other cases are represented as str.
+                It is useful to mark word as unchangeable which is already in phoneme representation.
+            apply_to_oov_word: Function that will be applied to out of phoneme_dict word.
+            ignore_ambiguous_words: Whether to not handle word via phoneme_dict with ambiguous phoneme sequences. Defaults to True.
+            heteronyms (str, Path, List): Path to file with heteronyms (every line is new word) or list of words.
+            encoding: Encoding type.
+            phoneme_probability (Optional[float]): The probability (0.<var<1.) that each word is phonemized. Defaults to None which is the same as 1.
+                Note that this code path is only run if the word can be phonemized. For example: If the word does not have a entry in the g2p dict, it will be returned
+                as characters. If the word has multiple entries and ignore_ambiguous_words is True, it will be returned as characters.
```

Comment:
`heteronyms`, `encoding` and `phoneme_probability` params are not used as inputs to `__init__`. Is there any reason why we decide to remove them in `self.__init__()`?

IIUC, after conversion into phonemes from Chinese characters, the heteronyms should not exist, so that `phoneme_probability` is always 1.0. So maybe that's why you remove them?

Reply:
A general question: how to address the case that OOV characters do not exist in pypinyin?

Reply:
Yes, I have no consider for polyphone disambiguation in this PR. Maybe it could be done through a unified frontend along with a prosody bert. 

For OOV cases in Chinese, I think pypinyin cover almost all of it. (Chinese characters are standord fixed limited set) In case it can't handle it, it would return the token unchanged. In the following ChinesePhoneTokenier, the out of dict token would be ignored with a warning information.

Reply:
sounds good!